### ISSUE TITLE
Getting KeyError: 1478 if the underlying connection is reconstructed.

### ISSUE THREAD

**Issue by atuljangra** [pika/pika#671]
Steps to reproduce:
1. Create a connection.
2. Create a channel and start pushing objects to a queue through that channel.
3. Process is sleeping/waiting when the underlying connection is closed.
4. Reestablish the connection.
5. Using the previous channel throws this error.

I think that we should throw an exception in such cases. The case is obviously flowed, but handling it as an exception makes more sense.

**Comment by gmr** [comment]
How are you reestablishing the connection?

**Comment by vitaly-krugl** [comment]
When a connection is closed, all channels on that connection are also (implicitly) closed. You cannot continue to use channels from a previously-closed connection.

**Comment by atuljangra** [comment]
Reestablishing the connection normally. Like you would create the normal one.
I understand that we should not use the channels from previously closed connection. But that should not be allowed by design.
Accessing a channel when the underlying connection is closed should throw a good descriptive exception.

**Comment by vitaly-krugl** [comment]
@atuljangra, please check against pika master branch. This issue should be resolved via PR #685, which addressed a lot of issues related to server-initiated closing of connections.

```ipython
In [1]: import pika

In [2]: c = pika.BlockingConnection()

In [3]: ch = c.channel()

After creating the channel, I used the RabbitMQ Management Web UI to force the connection closed

In [4]: c.process_data_events()
ConnectionClosed: (320, 'CONNECTION_FORCED - Closed via management plugin')

In [5]: c.process_data_events()
ConnectionClosed

In [6]: ch.is_open
Out[6]: False

In [7]: ch.queue_delete("abcdef")
ChannelClosed


```

**Comment by atuljangra** [comment]
Yes. This should resolve the issue.
Thanks @vitaly-krugl 
Closing.
